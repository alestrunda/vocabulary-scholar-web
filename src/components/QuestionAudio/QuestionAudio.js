import React from 'react';
import PropTypes from 'prop-types';
import AudioIconApiLoad from '../../containers/AudioIconApiLoad';

const QuestionAudio = ({
  audioSrc,
  className,
  onQuestionReady,
  style,
  wordID,
}) => {
  const handleQuestionReady = () => {
    onQuestionReady && onQuestionReady();
  };

  return (
    <div className={className} style={style}>
      <AudioIconApiLoad
        onReady={handleQuestionReady}
        size={100}
        sizeIcon={60}
        wordID={wordID}
        src={audioSrc}
        hideIfNoSource={false}
        autoplay
      />
    </div>
  );
};

QuestionAudio.propTypes = {
  className: PropTypes.string,
  style: PropTypes.object,
  audioSrc: PropTypes.string,
  wordID: PropTypes.string,
  onQuestionReady: PropTypes.func,
};

export default QuestionAudio;
