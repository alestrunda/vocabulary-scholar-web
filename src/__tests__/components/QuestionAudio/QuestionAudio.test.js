import React from 'react';
import { shallow } from 'enzyme';

import QuestionAudio from '../../../components/QuestionAudio';

it('renders correctly', () => {
  const root = shallow(<QuestionAudio audioSrc="file.mp3" wordID="school" />);
  expect(root).toMatchSnapshot();
});
