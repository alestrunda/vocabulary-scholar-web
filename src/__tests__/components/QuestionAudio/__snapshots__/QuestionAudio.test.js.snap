// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <QuestionAudio
    audioSrc="file.mp3"
    wordID="school"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <AudioIconApiLoad
        autoplay={true}
        hideIfNoSource={false}
        loadFromApiIfNotAvailable={true}
        onReady={[Function]}
        size={100}
        sizeIcon={60}
        src="file.mp3"
        wordID="school"
      />,
      "className": undefined,
      "style": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "autoplay": true,
        "hideIfNoSource": false,
        "loadFromApiIfNotAvailable": true,
        "onReady": [Function],
        "size": 100,
        "sizeIcon": 60,
        "src": "file.mp3",
        "wordID": "school",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <AudioIconApiLoad
          autoplay={true}
          hideIfNoSource={false}
          loadFromApiIfNotAvailable={true}
          onReady={[Function]}
          size={100}
          sizeIcon={60}
          src="file.mp3"
          wordID="school"
        />,
        "className": undefined,
        "style": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "autoplay": true,
          "hideIfNoSource": false,
          "loadFromApiIfNotAvailable": true,
          "onReady": [Function],
          "size": 100,
          "sizeIcon": 60,
          "src": "file.mp3",
          "wordID": "school",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
